### PR TITLE
implementation of showing movie list

### PR DESCRIPTION
**Pull request by briansetzesglobant**
Made the implementation of Show a list of the trending movies.

https://user-images.githubusercontent.com/84338449/122269152-45c5f080-ceb3-11eb-87fe-4b3b3af99644.mp4

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -1,113 +1,21 @@
...
-            Text(
-              'You have pushed the button this many times:',
-            ),
-            Text(
-              '$_counter',
-              style: Theme.of(context).textTheme.headline4,
-            ),
-          ],
+    return MaterialApp(
+      home: MoviesList(
```
Comment (FranciscoGaraventa) [inline lib/main.dart]: Use a more descriptive name to call this widget. You can use HomePage or something like that.

File: lib/src/blocs/movies_bloc.dart [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import 'dart:async';
+import '../models/item_model.dart';
+import '../resources/movies_repository.dart';
+import 'i_movies_bloc.dart';
+
+class MoviesBloc extends IMoviesBloc {
+  MoviesRepository moviesRepository = MoviesRepository();
+  StreamController<ItemModel> moviesStreamController = StreamController();
```
Comment (FranciscoGaraventa) [inline lib/src/blocs/movies_bloc.dart]: This variables are private inside the Bloc so you need to add at beginning of each an '_'

File: lib/src/blocs/movies_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    this.moviesStreamController.close();
+  }
+
+  @override
+  Stream<ItemModel> get moviesStream => this.moviesStreamController.stream;
+
+  @override
+  void fetchAllMovies() async {
+    final _moviesList = await moviesRepository.fetchAllMovies();
+    this.moviesStreamController.sink.add(_moviesList,);
```
Comment (FranciscoGaraventa) [inline lib/src/blocs/movies_bloc.dart]: The comma symbol after _movieList is not necessary

File: lib/src/models/item_model.dart [hunk 4]
```diff
@@ -0,0 +1,90 @@
+class ItemModel {
```
Comment (FranciscoGaraventa) [inline lib/src/models/item_model.dart]: This class name can be more descriptive, like MovieModel or something like that

File: lib/src/blocs/movies_bloc.dart [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+class MoviesBloc extends IMoviesBloc {
+  MoviesRepository moviesRepository = MoviesRepository();
+  StreamController<ItemModel> moviesStreamController = StreamController();
+
+  @override
+  void initialize() {}
+
+  @override
+  void dispose() {
+    this.moviesStreamController.close();
```
Comment (FranciscoGaraventa) [inline lib/src/blocs/movies_bloc.dart]: Using "this" is the same as not using it. Remove it of all this file.

File: lib/src/models/item_model.dart [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+  String get title => _title;
+
+  double get vote_average => _vote_average;
+
+  bool get video => _video;
+
+  int get id => _id;
+
+  int get vote_count => _vote_count;
+}
```
Comment (FranciscoGaraventa) [inline lib/src/models/item_model.dart]: Move this class into another file and like ItemModel comment change this to a more descriptive class name

File: lib/src/ui/movies_list.dart [hunk 7]
```diff
@@ -0,0 +1,89 @@
+import 'package:flutter/material.dart';
+import '../../utils/constants.dart';
+import '../blocs/i_movies_bloc.dart';
+import '../models/item_model.dart';
+
+class MoviesList extends StatefulWidget {
```
Comment (FranciscoGaraventa) [inline lib/src/ui/movies_list.dart]: Change class name to more descriptive.

File: lib/src/ui/movies_list.dart [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+  void dispose() {
+    widget.bloc.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(widget.title),
```
Comment (FranciscoGaraventa) [inline lib/src/ui/movies_list.dart]: You missed a trailing comma here

File: lib/src/ui/movies_list.dart [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+                Constants.paddingNumber,
+              ),
+              child: Image.network(
+                Constants.imageNetwork +
+                    snapshot.data.results[index].poster_path,
+                fit: BoxFit.cover,
+              ),
+            ),
+          );
+        });
```
Comment (FranciscoGaraventa) [inline lib/src/ui/movies_list.dart]: You missed a trailing comma here

File: test/widget_test.dart [hunk 10]
```diff
@@ -13,7 +13,7 @@ import 'package:movies/main.dart';
 void main() {
   testWidgets('Counter increments smoke test', (WidgetTester tester) async {
     // Build our app and trigger a frame.
-    await tester.pumpWidget(MyApp());
+    //await tester.pumpWidget(MyApp());
```
Comment (FranciscoGaraventa) [inline test/widget_test.dart]: Is not necessary to comment this, change MyApp to App

File: lib/utils/constants.dart [hunk 11]
```diff
@@ -0,0 +1,11 @@
+abstract class Constants {
+  static const int crossAxisCountNumber = 2;
+  static const double paddingNumber = 10.0;
+  static const String imageNetwork =
+      "https://image.tmdb.org/t/p/w185";
+
+  static const String uriClient =
+      "https://api.themoviedb.org/3/movie/popular?api_key=385737c6d715026a1cd89526ff9d8d44";
+  static const int status = 200;
+  static const String error = 'I failed the load';
```
Comment (FranciscoGaraventa) [inline lib/utils/constants.dart]: This error message can be: 'An error has ocurred while loading movies"

**Review by MarcosCGlobant — COMMENTED** [review]
Approved after Frans comments

File: lib/main.dart (hunk 1 above)
Comment (briansetzesglobant) [reply lib/main.dart]: fixed

File: lib/src/blocs/movies_bloc.dart (hunk 2 above)
Comment (briansetzesglobant) [reply lib/src/blocs/movies_bloc.dart]: fixed

File: lib/src/blocs/movies_bloc.dart (hunk 3 above)
Comment (briansetzesglobant) [reply lib/src/blocs/movies_bloc.dart]: fixed

File: lib/src/models/item_model.dart (hunk 4 above)
Comment (briansetzesglobant) [reply lib/src/models/item_model.dart]: done

File: lib/src/blocs/movies_bloc.dart (hunk 5 above)
Comment (briansetzesglobant) [reply lib/src/blocs/movies_bloc.dart]: done

File: lib/src/models/item_model.dart (hunk 6 above)
Comment (briansetzesglobant) [reply lib/src/models/item_model.dart]: done

File: lib/src/ui/movies_list.dart (hunk 7 above)
Comment (briansetzesglobant) [reply lib/src/ui/movies_list.dart]: done

File: lib/src/ui/movies_list.dart (hunk 8 above)
Comment (briansetzesglobant) [reply lib/src/ui/movies_list.dart]: done

File: test/widget_test.dart (hunk 10 above)
Comment (briansetzesglobant) [reply test/widget_test.dart]: fixed

File: lib/src/ui/movies_list.dart (hunk 9 above)
Comment (briansetzesglobant) [reply lib/src/ui/movies_list.dart]: fixed

File: lib/utils/constants.dart (hunk 11 above)
Comment (briansetzesglobant) [reply lib/utils/constants.dart]: done

**Review by FranciscoGaraventa — APPROVED** [review]
LGTM

**Review by MarcosCGlobant — APPROVED** [review]
LGTM

**Pull request merged**